### PR TITLE
langchain-openai: openai proxy added to base embeddings

### PR DESCRIPTION
**Pull request by PabloKarpacho**
- [ ] **PR title**: "langchain-openai: openai proxy added to base embeddings"

- [ ] **PR message**: 
    - **Description:** 
    Dear langchain developers,
    You've already supported proxy for ChatOpenAI implementation in your package. At the same time, if somebody needed to use proxy for chat, it also could be necessary to be able to use it for OpenAIEmbeddings. That's why I think it's important to add proxy support for OpenAI embeddings. That's what I've done in this PR.

@baskaryan

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -91,3 +91,6 @@ extend-exclude = [
 # Don't require them to change at once, though we should look into them eventually.
 "cookbook/gymnasium_agent_simulation.ipynb" = ["F821"]
 "docs/docs/integrations/document_loaders/tensorflow_datasets.ipynb" = ["F821"]
+
+[tool.setuptools]
+py-modules = []
```
Comment (eyurtsev) [inline pyproject.toml:96]: Could you remove this change?

Comment (PabloKarpacho) [reply pyproject.toml]: sure, removed

**Review by eyurtsev — APPROVED** [review]

File: libs/partners/openai/langchain_openai/embeddings/base.py [hunk 2]
```diff
@@ -282,12 +282,32 @@ def validate_environment(cls, values: Dict) -> Dict:
             "default_headers": values["default_headers"],
             "default_query": values["default_query"],
         }
+        
+        openai_proxy = values["openai_proxy"]
         if not values.get("client"):
+            if openai_proxy and not values["http_client"]:
```
Comment (eyurtsev) [inline libs/partners/openai/langchain_openai/embeddings/base.py:288]: @baskaryan what pattern do we use here if openai_proxy and http_client have been co-specified? Do we usually raise or issue a warning?

Comment (baskaryan) [reply libs/partners/openai/langchain_openai/embeddings/base.py:288]: raise a warning if openai_proxy and either of http_client or http_async_client is specified

**Comment by PabloKarpacho** [comment]
Will changes be merged in master's branch, or I need to do smth else?

File: libs/partners/openai/langchain_openai/embeddings/base.py (hunk 2 above)
Comment (baskaryan) [reply libs/partners/openai/langchain_openai/embeddings/base.py:288]: @PabloKarpacho could we update to raise warning in above case ^

Comment (PabloKarpacho) [reply libs/partners/openai/langchain_openai/embeddings/base.py:288]: Sure, do you have example in your code&

**Review by baskaryan — APPROVED** [review]

**Pull request merged**